docker/daemon/client.go

```diff
@@ -0,0 +1,73 @@
+package daemon
+
+import (
+	"github.com/containers/image/types"
+	dockerclient "github.com/docker/docker/client"
+	"github.com/docker/go-connections/tlsconfig"
+	"net/http"
+	"path/filepath"
+)
+
+const (
+	// The default API version to be used in case none is explicitly specified
+	defaultAPIVersion = "1.22"
+)
+
+// NewDockerClient initializes a new API client based on the passed SystemContext.
+func newDockerClient(ctx *types.SystemContext) (*dockerclient.Client, error) {
+	if ctx == nil {
+		return dockerclient.NewClient(dockerclient.DefaultDockerHost, defaultAPIVersion, nil, nil)
+	}
+
+	httpClient, err := tlsConfig(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	host := ctx.DockerDaemonHost
+	if host == "" {
+		host = dockerclient.DefaultDockerHost
+	}
+
+	version := ctx.DockerDaemonAPIVersion
+	if version == "" {
+		version = defaultAPIVersion
+	}
+
+	cli, err := dockerclient.NewClient(host, version, httpClient, nil)
+	if err != nil {
+		return cli, err
+	}
+
+	return cli, nil
+}
+
+func tlsConfig(ctx *types.SystemContext) (*http.Client, error) {
+	if ctx == nil {
```

Comment:
While dropping the `DockerDaemonCertPath` from here is an improvement, this is never true because `newDockerClient` has a similar check.

Reply:
true, but imo the function needs to be consistent in itself and is responsible for its own verification of parameters. Who says that  _newDockerClient_ will always be the only caller if this method.

Reply:
so what's about I drop the 'if ctx == nil' case? In this case, there will only be a single path. _ newDockerClient_ is the only caller and it does already the _nil_ check as you pointed out.

Reply:
No, there would still be two paths: the `newDockerClient` `if ctx == nil` path (with `nil` `httpClient`, i.e. `*FromEnvironment` called), and the `tlsConfig` path.